config/src/main/java/com/epam/aidial/core/config/AccessControlled.java

```diff
@@ -0,0 +1,29 @@
+package com.epam.aidial.core.config;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import lombok.Data;
+
+import java.util.Collection;
+import java.util.Set;
+
+@Data
+public abstract class AccessControlled {
+
+    private String name;
+
+    @JsonAlias({"userRoles", "user_roles"})
+    private Set<String> userRoles;
+
+    public boolean hasAccess(Collection<String> roles) {
+        if (userRoles == null) {
+            return true;
+        }
+
+        if (userRoles.isEmpty() || roles == null || roles.isEmpty()) {
```

Comment:
Could you please keep the same code as it was? It looks more compact.

```
if (expectedUserRoles == null) {
             return true;
         }

         return !expectedUserRoles.isEmpty()
                 && actualUserRoles.stream().anyMatch(expectedUserRoles::contains);
```

Reply:
returned the previous logic